src/Microsoft.AspNet.SignalR.Redis/RedisMessageBus.cs

```diff
@@ -74,14 +77,41 @@ public virtual void OpenStream(int streamIndex)
 
         protected override Task Send(int streamIndex, IList<Message> messages)
         {
-            return _connection.ScriptEvaluateAsync(
+            string GetMessageContent(Message message)
+            {
+                var content = Encoding.UTF8.GetString(message.Value.Array, message.Value.Offset, message.Value.Count);
+                return $"Key:{message.Key},ID:{message.MappingId},Content:{content}";
+            }
+
+            async Task WaitForAndTraceResult(Task<RedisResult> task)
+            {
+                var result = await task;
+                var values = (RedisValue[]) result;
+                _trace.TraceEvent(TraceEventType.Verbose, 0, $"Published message with key: {(int)values[0]}");
+            }
+
+            if (_trace.Switch.ShouldTrace(TraceEventType.Verbose))
+            {
+                var messageContent = "[" + string.Join(";", messages.Select(m => GetMessageContent(m))) + "]";
+                _trace.TraceEvent(TraceEventType.Verbose, 0, "Publishing message: {0}", messageContent);
+            }
+
+            var execTask = _connection.ScriptEvaluateAsync(
                 _db,
                 @"local newId = redis.call('INCR', KEYS[1])
                   local payload = newId .. ' ' .. ARGV[1]
                   redis.call('PUBLISH', KEYS[1], payload)
                   return {newId, ARGV[1], payload}",
                 _key,
                 RedisMessage.ToBytes(messages));
+
+            // Sketchy AF, but we know that our RedisConnection returns the Task<RedisResult> as the Task, so we can downcast
+            if (_trace.Switch.ShouldTrace(TraceEventType.Verbose) && execTask is Task<RedisResult> resultTask)
```

Comment:
Why not do this in ScriptEvaluateAsync then?

Reply:
Well, we don't know what the script is doing in `ScriptEvaluateAsync`, without parsing the Lua or matching it against a known script, which seems just as gross...